### PR TITLE
Implemented browser info

### PR DESCRIPTION
**Pull request by tymees**
This PR implements #12 

This is done by adding a new submodule, browser. This module uses UAParser to get all info from the browser's user agent, except for resulution info. 

Aditionally, the existing ``isMobile``, ``isMobileOrTablet`` and ``isTouchCapable`` have been moved to this new submodule. ``isMobile``, ``isMobileOrTablet`` also have been modified to use UAParser where possible, as this will give more accurate results. As UAParser is loaded asynchronicely, an experiment might access these methods before UAParser is setup. Thus, the old implementation is still used as a fallback on these methods. 
For backwards compatibility, all three old methods under the ``uil`` namspace are now aliasses to the new methods in the ``uil.browser`` namespace. (And marked as deprecated). 

New methods:
``getBrowser``: Returns the browser name and version
``getDevice``: Returns the device name, or "Unknown Device" if it can't be identified
``getOS``: Returns the OS name and version (if present)
``isTablet``: returns if this device is a tablet. If UAParser hasn't been loaded yet, some mobile devices will also be seen as tablets.
``getBrowserInfo``: Returns data from ``getBrowser``, ``getDevice``, ``getOS``, ``isMobile``, ``isTablet`` and ``isTouchCapable`` as a dict. 
``getResolutionInfo``: Returns a lot of resolution information. Please see the docstring for a better description. 

Lastly, submodule loading now also adds an ``id`` field. This way submodules can use an onload on their dependencies directly, to wait till they are loaded. Using a regular ``window.onload`` won't work, as this will be executed after the onload of the experiment.

### CODE REVIEW

File: jspsych-uil-utils.js [hunk 1]
```diff
@@ -310,26 +314,20 @@ var uil = {};
...
-        })(navigator.userAgent || navigator.vendor || window.opera);
-        return check;
+        return context.browser.isMobile();
```
Comment (maartenuni) [inline jspsych-uil-utils.js:321]: Warn user of use of deprecated functions, so  something like:

```JavaScript
deprecatedFunctionWarning = function (old, solution) {
    console.warn(`Use of decprecated function ${old}, solution: ${solution}"`);
}
context.isMobile = function () {
    deprecatedFunctionWarning("uil.isMobile()", "use uil.browser.isMobile() instead";
    return context.browser.isMobile();
}

File: jspsych-uil-utils.js [hunk 2]
```diff
@@ -310,26 +314,20 @@ var uil = {};
...
-        })(navigator.userAgent || navigator.vendor || window.opera);
-        return check;
+        return context.browser.isMobileOrTablet();
```
Comment (maartenuni) [inline jspsych-uil-utils.js:330]: See comment on context.isMobile()

File: jspsych-uil-utils.js [hunk 3]
```diff
@@ -343,30 +341,11 @@ var uil = {};
...
-            );
-            if (media_query && media_query.media === "(pointer:coarse)") {
-                // Double negation to cast media_query.matches to a bool
-                // (I think)
-                has_touchscreen = !!media_query.matches;
-            }
-        }
-
-        return has_touchscreen
+        return context.browser.isTouchCapable();
```
Comment (maartenuni) [inline jspsych-uil-utils.js:348]: See comment on context.isMobile()

File: jspsych-uil-utils.js [hunk 4]
```diff
@@ -51,7 +51,11 @@ var uil = {};
     const CONTENT_TYPE = 'Content-Type';
     const CONTENT_TYPE_TEXT_PLAIN = 'text/plain';
 
-    const LIBRARIES = ['jspsych-uil-randomization.js']
+    const LIBRARIES = [
+        'jspsych-uil-randomization.js',
+        'libs/ua-parser.min.js', // Dependency of uil-browser
```
Comment (maartenuni) [inline jspsych-uil-utils.js:56]: If possible use 'jspsych-uil-browser' to load this library onto the uil context. I would like to keep a list of public libraries here. Alternatively, create a
```JavaScript
const PRIVATE_LIBS = ['libs/ua-parser.min.js'];
```
And load the private libraries/dependencies separately.

**Comment by maartenuni** [comment]
Thanks! Is looking very good and easy to use! Perhaps we should document that the browsersize is variable... We cannot control whether the participant decides to unfullscreen the experiment or changes the browser window size.

File: jspsych-uil-utils.js (hunk 4 above)
Comment (tymees) [reply jspsych-uil-utils.js]: Fixed in 8c6411edb6e59a20997a75f313d9dabbf3fa29fe

I've decided to use your second suggestion, as this keeps loading dependencies somewhat centralised. I think this is preferred, to keep better track of what is loaded when. (Loading orders are annoying enough to debug).
In addition, placing it in a submodule would either require code duplication or exposing private methods of ``uil``, which is less preferable in my opinion

File: jspsych-uil-utils.js (hunk 1 above)
Comment (tymees) [reply jspsych-uil-utils.js:325]: Added in 23f126770b38f5400a2e74149eaf6b58f1efcfcb

I didn't use a helper function, adding ~5 lines (documentation included) just to do a console.warn is, in my opinion, a bit overkill. If you disagree, I'll add it anyway :)

**Comment by tymees** [comment]
Good idea, I've added some documentation in f5dbfc6b8b4341bcf50b1111fd08671fe21c221b

I think I've addressed all your points with these 4 commits?

**Comment by maartenuni** [comment]
Lijkt mij prima, wat mij betreft kan je hem mergen

**Pull request merged**